File: contributions/ruby/company/subunit.rb
class Subunit
  
  attr_reader :pu, :du
  
  
  def pu=(pu)
    @pu = pu
    @du = nil
  end
  
  def du=(du)
    @pu = nil
    @du = du
  end
  
  def total
    ttl = 0.0
    if (pu != nil)
      ttl = pu.total
    else
      ttl = du.total
    end
    ttl
  end
  
  def cut
    if (pu!= nil)
      pu.cut
    else
      du.cut
    end
  end
  
end

File: contributions/strutsXml/src/main/java/org/softlang/model/Subunit.java
package org.softlang.model;

import java.io.Serializable;

/**
 * A subunit is either a person unit or a department unit
 * 
 */
public abstract class Subunit implements Serializable {

	private static final long serialVersionUID = -2008895922137165250L;

	public abstract double total();
	public abstract void cut();
	public boolean isDepartment() { 
		return false;
	}
}
